spec/unit/lib/cloud_controller/check_stacks_spec.rb

```diff
@@ -0,0 +1,122 @@
+require 'spec_helper'
+require 'yaml'
+require 'cloud_controller/check_stacks'
+require 'tasks/rake_config'
+
+module VCAP::CloudController
+  RSpec.describe CheckStacks do
+    let(:stack_file_contents) do
+      {
+        'default' => 'cflinuxfs4',
+        'stacks' => [
+          cflinuxfs4
+        ],
+        'deprecated_stacks' => 'cflinuxfs3'
+      }
+    end
+
+    let(:cflinuxfs3) { { 'name' => 'cflinuxfs3', 'description' => 'fs3' } }
+    let(:cflinuxfs4) { { 'name' => 'cflinuxfs4', 'description' => 'fs4' } }
+
+    before do
+      # binding.pry
+      Stack.dataset.destroy
+      file = Tempfile.new
+      file.write(stack_file_contents.to_yaml)
+      file.close
+      Stack.configure(file)
+      Stack.populate
+      TestConfig.override(stacks_file: file.path)
+    end
+
+    let(:stack_checker) { CheckStacks.new(TestConfig.config_instance) }
+
+    describe 'the deprecated stacks is nil' do
+      let(:stack_file_contents) do
+        {
+          'default' => 'cflinuxfs4',
+          'stacks' => [
+            cflinuxfs3,
+            cflinuxfs4
+          ]
+        }
+      end
+
+      it 'does nothing' do
+        expect { stack_checker.validate_stacks }.not_to raise_error
+      end
+    end
+
+    describe 'there are no deprecated stacks' do
+      let(:stack_file_contents) do
+        {
+          'default' => 'cflinuxfs4',
+          'stacks' => [
+            cflinuxfs3,
+            cflinuxfs4
+          ],
+          'deprecated_stacks' => []
+        }
+      end
+
+      it 'does nothing' do
+        expect { stack_checker.validate_stacks }.not_to raise_error
+      end
+    end
+
+    describe 'the deprecated stack is in the config' do
+      let(:stack_file_contents) do
+        {
+          'default' => 'cflinuxfs4',
+          'stacks' => [
+            cflinuxfs3,
+            cflinuxfs4
+          ],
+          'deprecated_stacks' => ['cflinuxfs3']
+        }
+      end
+
+      describe 'the deprecated stack is in the db' do
```

Comment:
*question* Do we need these describes for the db behaviour? Looking at the code, if the deprecated stack is in the stacks list we don't even check the DB. We return early here https://github.com/cloudfoundry/cloud_controller_ng/pull/3894/files#diff-b51e853460880f38ea614d45c335bfdaa98ebf81f6166ce4c13ffab95bfbf871R23